### PR TITLE
Use maximize-build-space for GH action

### LINKED ISSUE #3244: No Space Left on Device for GitHub Action
Zoltar Upload has been failing since Friday with the error indicating we're at limit for device storage in GitHub Action

https://github.com/reichlab/covid19-forecast-hub/runs/2142306517?check_suite_focus=true

- [x] Temporary Fix: Remove the steps: checking actions/setup-node@v1 and npm install
- [x] Temporary Fix: remove preinstalled software from GitHub [PR](https://github.com/reichlab/covid19-forecast-hub/pull/3253)
- [ ] Long-term: use of this GH action https://github.com/easimon/maximize-build-space, which can save up to 40 GBs

### PR DESCRIPTION
**Pull request by Khoale1096**
## Description

Use the market place maximize-build-space github action to free up ~40 GBs of space on GH action virtual environment. Closes #3244 

---
> _If this section does not apply to you, please delete it in your PR_  

If this is a **new team submission**, please include the following details :-  
- Team name: 
- Model Name: 
- Institution:  
---

> _If this section does not apply to you, please delete it in your PR_  

If you are **adding new forecasts** to an existing model, please include the following details:- 
- Team name: 
- Model name that is being updated:  
---

## Checklist

- [ ] Specify a proper PR title with your team name.
- [ ] All validation checks ran successfully on your branch. Instructions to run the tests locally is present [here](https://github.com/reichlab/covid19-forecast-hub/wiki/Running-Checks-Locally).

### CODE REVIEW

**Review by hannanabdul55 — APPROVED** [review]
Sounds good! @Khoale1096 - can we keep the `df --human-readable` script so that we get to see how much free space is there?

**Review by ydhuang28 — CHANGES_REQUESTED** [review]
Could we add back the `df --human-readable` command after the `maximize-build-space` action?

File: .github/workflows/upload_to_zoltar.yml [hunk 1]
```diff
@@ -15,19 +15,17 @@ jobs:
...
-    - name: Free disk space
-      run: |
-          df --human-readable
-          sudo apt clean
-          docker rmi $(docker image ls --all --quiet)
-          rm --recursive --force "$AGENT_TOOLSDIRECTORY" 
-    - run: sudo apt-get remove ghc-** azure-cli firefox google-cloud-sdk google-chrome-stable hhvm dotnet-sdk-** dotnet-runtime-** moby-** adoptopenjdk-** mongodb-org-** mysql-**
-    - run: sudo apt-get autoremove
-    - name: Check space available
-      run: df --human-readable
```
[the remark this answers is not in the document]
Comment (Khoale1096) [reply .github/workflows/upload_to_zoltar.yml:27]: @hannanabdul55 Yep sorry I deleted that part by accident

**Review by ydhuang28 — APPROVED** [review]
Looks good!

**Pull request merged**